src/realm/cluster.cpp

```diff
@@ -1873,6 +1905,14 @@ Obj ClusterTree::insert(ObjKey k, const FieldValues& values)
                         index->insert(k, init_value.get<Timestamp>());
                     }
                     break;
+                case col_type_Decimal:
+                    if (init_value.is_null()) {
+                        index->insert(k, ArrayDecimal128::default_value(nullable));
+                    }
+                    else {
+                        index->insert(k, init_value.get<Timestamp>());
```

Comment:
wrong type here (copy-pasta?)

Reply:
Definitely wrong. We do not support indexing Decimal128 (yet - as we do not support indexing float and double). Changed the implementation to add support for ObjectId, which was missing. Hope everybody can live with this.